src/bridge/cockpitrouter.c

```diff
@@ -1631,6 +1665,102 @@ cockpit_router_dbus_startup (CockpitRouter *router)
     }
 }
 
+/* Superuser init */
+
+static void superuser_init_step (CockpitRouter *router);
+
+static void
+superuser_init_start (CockpitRouter *router,
+                      const gchar *id,
+                      const gchar *password)
+{
+  router->superuser_init_in_progress = TRUE;
+  router->superuser_init_next_rule = router->rules;
+  router->superuser_init_id = g_strdup (id);
+  router->superuser_init_password = g_strdup (password);
+
+  superuser_init_step (router);
+}
+
+static void
+superuser_init_step_done (const gchar *error, gpointer user_data)
+{
+  CockpitRouter *router = user_data;
+
+  if (error)
+    {
+      router->superuser_rule = NULL;
+      superuser_init_step (router);
+    }
+  else
+    {
+      router->superuser_init_in_progress = FALSE;
+      g_free (router->superuser_init_id);
+      router->superuser_init_id = NULL;
+      g_free (router->superuser_init_password);
```

Comment:
Please also overwrite the password before freeing it, so that it doesn't remain in memory. As it's also contained in the init message, the same mlock+clear+zero copy should be done to it.